suite-native/module-send/src/hooks/useAddressValidationAlerts.tsx

```diff
@@ -0,0 +1,138 @@
+import { G } from '@mobily/ts-belt';
+
+import { SendStackParamList, SendStackRoutes } from '@suite-native/navigation';
+import { Translation } from '@suite-native/intl';
+import { selectAccountTokenSymbol, TokensRootState } from '@suite-native/tokens';
+import { useAlert } from '@suite-native/alerts';
+import { useFormContext } from '@suite-native/forms';
+import { isAddressValid } from '@suite-common/wallet-utils';
+import { AccountsRootState, selectAccountNetworkSymbol } from '@suite-common/wallet-core';
+import TrezorConnect from '@trezor/connect';
+import { Link } from '@suite-native/link';
+
+import { getOutputFieldName } from '../utils';
+import { TokenOfNetworkAlertBody } from '../components/TokenOfNetworkAlertContent';
+
+type UseAddressValidationAlertsArgs = {
+    inputIndex: number;
+};
+
+const CHECKSUM_LINK_URL = 'https://trezor.io/learn/a/evm-address-checksum-in-trezor-suite';
+
+export const useAddressValidationAlerts = ({ inputIndex }: UseAddressValidationAlertsArgs) => {
+    const {
+        params: { tokenContract, accountKey },
+    } = useRoute<RouteProp<SendStackParamList, SendStackRoutes.SendOutputs>>();
+    const [wasAddressChecksummed, setWasAddressChecksummed] = useState(false);
+    const [wasTokenAlertDisplayed, setWasTokenAlertDisplayed] = useState(
+        G.isNullable(tokenContract),
+    );
+    const { showAlert } = useAlert();
+
+    const tokenSymbol = useSelector((state: TokensRootState) =>
+        selectAccountTokenSymbol(state, accountKey, tokenContract),
+    );
+    const networkSymbol = useSelector((state: AccountsRootState) =>
+        selectAccountNetworkSymbol(state, accountKey),
+    );
+
+    const { watch, setValue } = useFormContext();
+
+    const addressFieldName = getOutputFieldName(inputIndex, 'address');
+    const addressValue = watch(addressFieldName);
+
+    const isFilledValidAddress =
+        addressValue && networkSymbol && isAddressValid(addressValue, networkSymbol);
+
+    const convertAddressToChecksum = useCallback(() => {
+        setValue(addressFieldName, toChecksumAddress(addressValue), {
+            shouldValidate: true,
+        });
+        setWasAddressChecksummed(true);
+    }, [addressFieldName, addressValue, setValue]);
+
+    const handleAddressChecksum = useCallback(async () => {
+        if (isFilledValidAddress && !checkAddressCheckSum(addressValue)) {
+            const params = {
+                descriptor: addressValue,
+                coin: networkSymbol,
+            };
+
+            const addressInfo = await TrezorConnect.getAccountInfo(params);
+
+            if (addressInfo.success) {
+                // Already used addresses are checksumed without displaying the alert.
+                const isUsedAddress = addressInfo.payload.history.total !== 0;
+                if (isUsedAddress) {
+                    convertAddressToChecksum();
+
+                    return;
+                }
+            }
+
+            showAlert({
```

Comment:
if call to `getAccountInfo` fails, it shows the alert. We maybe dont want this, whyt?

Reply:
Good thinking. But this is a design requirement. If we are unable to get information if the address was user or not, the bottom sheet should be displayed. It works the same in desktop implementation. The only difference is that they do not use a bottom sheet but a banner instead.